### ISSUE TITLE
[TW-1070] Comply XDG Base Directory Specification

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1105]
_Gleb Javorsky on 2011-01-05T08:29:51Z says:_ 

 It would be nice if Taskwarrior comply this specification:
http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html

then ~/.task/ appears to be ~/.local/share/task/ or ~/.cache/task/
~/.taskrc can be in ~/.config/task/config
and so on

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2011-01-05T08:29:51Z
Modified: 2014-02-09T02:16:58Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2011-01-09T14:25:13Z says:_ 

 Reasons:
*  No benefit to taskwarrior through compliance
*  Never heard of it

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2011-01-11T14:43:17Z says:_ 

 From the author:

<pre>
just want you to 
know, that awesome, luakit, gajim, chromium, midori, 
vlc and etc using that specification. and for me, like 
for a regular user is very handy to store all my 
configs in one directory ~/.config
</pre>

**Comment by taskwarrior** [comment]
_Gleb Javorsky on 2011-01-17T05:46:16Z says:_ 

 At least you could make so, that taskwarrior will search config file in ~/.config/task/config, if can't find it in ~/.taskrc. It will be enought.

**Comment by taskwarrior** [comment]
_Cory Donnelly on 2011-01-17T10:44:38Z says:_ 

 At first I thought the XDG Base Directory specification was a good idea -- I like the idea of enabling users to reduce cruft in $HOME -- but I don't see a lot of utility otherwise.

Gleb Javorsky wrote:
> At least you could make so, that taskwarrior will search config file in ~/.config/task/config, if can't find it in ~/.taskrc. It will be enought.

I can't speak for anyone else, but I think implementing 1/6 of an emerging standard is worse than nothing at all.

**Comment by tbabej** [comment]
Support for taskrc location defined by `XDG_CONFIG_HOME` implemented in #2316. The other data locations can be then defined in the taskrc file itself.

Coincidentally same sort of middle ground Gleb Javorsky suggested almost a decade ago :slightly_smiling_face:
